tooling/greybox_fuzzer/src/mutation/array.rs

```diff
@@ -0,0 +1,201 @@
+            // Pick the length of the sequence from 1 to maximum available
+            let sequence_length = self.prng.gen_range(1..=(buffer_length - index));
+
+            let source_position = self.prng.gen_range(0..=(buffer_length - sequence_length));
+            // If first buffer is selected for the chunk, do nothing (we already have that part in the result)
+            // If the second is selected, copy the chunk into result
+            match BASIC_SPLICE_CANDIDATE_PRIORITIZATION_CONFIGURATION.select(self.prng) {
+                SpliceCandidate::First => {
+                    result.extend_from_slice(
+                        &first_buffer[source_position..(source_position + sequence_length)],
+                    );
+                }
+                SpliceCandidate::Second => {
+                    result.extend_from_slice(
+                        &second_buffer[source_position..(source_position + sequence_length)],
+                    );
+                }
+            }
+            index += sequence_length;
+        }
+        result
+    }
+
+    /// Generate a combination of two string inputs
+    pub fn splice(&mut self, first_input: &InputValue, second_input: &InputValue) -> InputValue {
+        let first_buffer = match first_input {
+            InputValue::Vec(inner_vec) => inner_vec,
+            _ => panic!("Shouldn't be used with other input value types"),
+        };
+
+        let second_buffer = match second_input {
+            InputValue::Vec(inner_vec) => inner_vec,
+            _ => panic!("Shouldn't be used with other input value types"),
+        };
+
+        assert!(!first_buffer.is_empty());
+        assert!(second_buffer.len() == first_buffer.len());
+
+        let result = match BASIC_SPLICE_MUTATION_CONFIGURATION.select(self.prng) {
+            SpliceMutation::PositionPreserving => {
+                self.structured_splice(first_buffer, second_buffer)
+            }
+            SpliceMutation::RandomChunks => self.chaotic_splice(first_buffer, second_buffer),
+        };
+        InputValue::Vec(result)
+    }
+
+    /// Perform one of structural mutations on the vector of input values
+    pub fn perform_structure_mutation_on_vector(
+        &mut self,
+        input_buffer: &Vec<InputValue>,
+    ) -> Vec<InputValue> {
+        match BASIC_VECTOR_STRUCTURE_MUTATION_CONFIGURATION.select(self.prng) {
+            StructuralMutation::ChaoticSelfSplice => {
+                self.chaotic_splice(input_buffer, input_buffer)
+            }
+            StructuralMutation::ChunkDuplication => self.duplicate_chunk(input_buffer),
+            StructuralMutation::Swap => self.swap(input_buffer),
+            StructuralMutation::RandomValueDuplication => {
+                panic!("Vector mutations should have a value duplication weight of zero")
+            }
+        }
+    }
+
+    /// Swap 2 random chunks in the buffer
+    fn swap(&mut self, buffer: &Vec<InputValue>) -> Vec<InputValue> {
```

Comment:
This method could be tested by having a separate function that receives all the values that are generated randomly (mainly to have a small proof that it works correctly).

Reply:
Added